framework/include/fvkernels/FVConstantScalarAdvection.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "FVKernel.h"
+
+template <ComputeStage compute_stage>
+class FVConstantScalarAdvection : public FVFluxKernel<compute_stage>
```

Comment:
I'm not sure we want to provide classes this specific in the framework - maybe we provide one that couples to a material property instead?

Reply:
The material version already exists. I think there is precedence to have classes like this in framework, e.g. Diffusion and not MatDiffusion. The argument is that it's easier to understand and get going. It could go under tests, though.

Reply:
Maybe we should follow the naming convention then - this is FVScalarAdvection and the other is FVScalarMatAdvection.  And do we really want a scalar advection? - would that be generally useful - or mabe we should just have vector advection and call it advection.

Reply:
Vector advection would require FV coupleable. Do we have that, and will it be any different than `Coupleable`?

Reply:
Your call, I'll change to what you think is best.

Reply:
Coupleable should work - haven't tested it though.  For this class - wouldn't you just make the Real be a RealVectorValue from the input parameters?  For nonlinear cases, we could couple to variables, but at least using the pronghorn approach we only care about coupling to materials (which can depend on other variables).

Reply:
Okay - this does treat velocity as a vector - what does the "scalar" in the name refer to?

Reply:
It transports a scalar quantity: energy, mass, concentration etc. So the scalar does not refer to `velocity` but to `u`.

Reply:
I see.  That seems a bit confusing since it isn't obvious what is scalar and moose uses scalar to refer to non-field variables.  Let's call this FVAdvection and then also have an FVMatAdvection.  That should keep things relatively consistent with Diffusion and MatDiffusion.  And for the vector case - when we need it later - we can add in "Vector" to the name.